### ISSUE TITLE
Bisq Lightning trade protocol

### ISSUE THREAD

**Issue by HenrikJannsen** [bisq-network/proposals#427]
The **Bisq Lightning trade protocol** enables that traders can use their off-chain Bitcoin on Lightning for trading Fiat currencies based on the Bisq Multisig protocol ported to the [Liquid](https://blockstream.com/liquid/) sidechain.

This trade protocol is intended to be implemented in Bisq2. Thanks to Roy Sheinfeld, the CEO of Breeze, who suggested the idea for the chained protocols.

## Overview
The protocol is a chain of 3 trades and 2 different protocols:
- First the traders swap their Lightning Bitcoin via a reverse Submarine swap to L-BTC (Submarine swap between Lightning and Liquid). This swap happens with any trader on the Bisq Submarine swap market, not with the trading peer.
- Then the trade happens on Liquid using the Bisq Multisig protocol (ported to Liquid).
- After that the L-BTC gets swapped back to Lightning by a Submarine swap (again with another trader/liquidity provider).

The traders do not need to have L-BTC. The trade execution on Liquid benefits from lower fees, faster confirmations and better privacy due confidential transactions. The Submarine swaps add further improvements for privacy as the coin merge of inputs in case of multiple trades is avoided.
The Submarine swap is an atomic swap and can be fully automated, thus makes it interesting for market makers and trading bots. In case there would not be enough liquidity on the Bisq Submarine swap market there is also the option to use centralized swap services like [Boltz](https://boltz.exchange/).

### Benefits:
- Allows users with Lightning Bitcoin to trade Fiat
- Compared to Bisq Multisig there are lower transaction fees, faster confirmations and better privacy.
- Better privacy due Lightning and submarine swaps (avoid coin merge issue)
- New markets for Submarine swap providers
- Reasonable good security for medium value trade amounts (similar security as with Bisq Multisig)

### Drawbacks:
- Higher complexity due chaining of 3 trades and 2 different protocols.
- Challenging for UX to hide complexity of the chained protocol
- The involvement of 2 additional traders for the Submarine swaps adds complexity and potential (rather small) risks.
- Both traders need to have BTC on Lightning for the security deposit.
- Liquid is a federated side chain and does not have the high level of censorship resistance as mainnet Bitcoin.
- Peg-in from mainnet Bitcoin to Liquid Bitcoin is trust-less, though for pegging-out L-BTC to BTC it requires authorization of the federation.
- The makers Bisq node need to be online as the take-offer process is interactive
- As the Fiat transfer is done on legacy systems like bank transfers it inherits all those drawbacks, like charge-back risk, slow transfer, privacy exposure to peer when the bank transfer details contain the real name. Though those various drawbacks can be mitigated by using payment methods which carry low charge-back risk, are fast or instant and use account IDs or email instead of real names. Certain payment methods avoid banking system completely like cash-by-mail or trading face to face.

## Roadmap
Bisq 2 is in its finals stage for release (release date estimated for Oct/Nov 2023). Initially it will only contain Bisq Easy (chat based protocol using reputation for security).
Once Bisq 2 is launched we will start to work on implementing the Bisq Multisig protocol (for mainnet Bitcoin).
Once Bisq Multisig protocol is out we can start working on porting it to Liquid and add it as an independent protocol.

In parallel the Submarine swap for BTC mainnet can get implemented and added as an independent protocol.
Once that is done the Submarine swap can be ported for Submarine swap to Liquid (L-BTC).

Once we have all the building blocks in place we can work on chaining those together to the Bisq Lightning trade protocol.

It would be great if 2 developer teams can split the work for the Multisig protocol and Submarine swap protocol parts and work in parallel.

## Requirements
For Liquid we have already started to integrate the [Elements](https://github.com/ElementsProject) wallet.
For Lightning we need to research for the best integration option. If anyone is interested to work on that, please get in touch.

**Comment by salokiam** [comment]
Great proposal, laying out the way to get Bisq into the Lightning world.

In the roadmap you mention
"Once Bisq 2 is launched we will start to work on implementing the Bisq Multisig protocol (for mainnet Bitcoin)."

1. Do we need the Bisq Multisig for the Lightning protocol?
2. How does it differ from the current Bisq1 multisig Protocol?
3. Is it a replacement for Bisq1 or will Bisq1 be in parallel?

**Comment by HenrikJannsen** [comment]
>     1. Do we need the Bisq Multisig for the Lightning protocol?

Yes. It we need to port the Bisq Multisig to Liquid first and that will be part of the chained Bisq Lightning protocol

>     2. How does it differ from the current Bisq1 multisig Protocol?

We started to work on the staged payouts for Bisq 1 Multisig already. That would have been a major improvement initially planned for the Bisq 2 version. Beside that we will see what else can be improved. We might get rid of the fee transactions as well. But require more analysis.

>     3. Is it a replacement for Bisq1 or will Bisq1 be in parallel?

Yes once the Bisq 2 Multisig (BTC mainnet) is out there is no reason to use Bisq 1 anymore. Will take a bit until users move over but focus will be on Bisq 2 and once the transition is completed (no more offers in Bisq1) we can retire Bisq 1 Multisig. BSQ swaps and DAO will be still a use case for Bisq 1 as it is not planned to implement the DAO in Bisq 2. BSQ swaps has low priority on the roadmap for Bisq 2.
